src/ui/src/core/index.ts

```diff
@@ -337,6 +394,12 @@ export function generateCore() {
 				}
 			} catch (error) {
 				logger.error("Error parsing WebSocket message:", error);
+				const { incrementMetric, METRIC_NAMES } = await import(
+					"@/observability/frontendMetrics"
+				);
```

Comment:
This seems repeating I think it is better to put it in a function and pass parameters of `incrementMetric` to that function, or import this at the top and don't use dynamic import. 
If you don't import dynamic, you don't need to make callbacks async.